Agents/GFAAgent/src/main/java/uk/ac/cam/cares/jps/agent/gfaagent/IntegrateFloors.java

```diff
@@ -0,0 +1,162 @@
+                    String polyIri = null;
+                    for (Map.Entry<String, List<Match<Document>>> entry : resultPoly.entrySet()) {
+                        for (Match<Document> match : entry.getValue()) {
+                            if(match.getScore().getResult()>polyScore && match.getScore().getResult()>0.5){
+                                System.out.println("Data: " + match.getData() + " Matched With: " + match.getMatchedWith() + " Score: " + match.getScore().getResult());
+                                polyScore = match.getScore().getResult();
+                                polyIri = match.getMatchedWith().getKey();
+                            }
+                        }
+                    }
+                    String pointIri = null;
+                    for (Map.Entry<String, List<Match<Document>>> entry : resultPoint.entrySet()) {
+                        for (Match<Document> match : entry.getValue()) {
+                            if(match.getScore().getResult()>pointScore && match.getScore().getResult()>0.5){
+                                System.out.println("Data: " + match.getData() + " Matched With: " + match.getMatchedWith() + " Score: " + match.getScore().getResult());
+                                pointScore = match.getScore().getResult();
+                                pointIri = match.getMatchedWith().getKey();
+                            }
+                        }
+                    }
+                    
+                    //store floors data based on building iri from osm agent
+                    int floors = hdbFloors.get(i).getFloors();
+                    String buildingiri = null;
+                    if(pointScore > polyScore && pointScore != 0) {
+                        buildingiri = pointIri;
+                    } else if (polyScore != 0){
+                        buildingiri = polyIri;
+                    }
+
+                    if (buildingiri != null) {
+                        String buildingSQLUpdate = "UPDATE building b SET storeys_above_ground = " + floors + 
+                                                " FROM cityobject_genericattrib cg\n" + 
+                                                "WHERE b.id = cg.cityobject_id AND cg.strval = '" + buildingiri + "';";
+                        postgisClient.executeUpdate(buildingSQLUpdate);
+                    }
+                    
+                }
+            }
+        } catch (SQLException e) {
+            throw new JPSRuntimeException("Error connecting to source database: " + e);
+        }       
+    }
+
+    private static final String polygonSQLQuery = "SELECT ogc_fid, addr_street, addr_housenumber, building_levels, building_iri FROM polygons WHERE addr_street IS NOT NULL OR addr_housenumber IS NOT NULL";
+    private static final String pointSQLQuery = "SELECT ogc_fid, addr_street, addr_housenumber, building_levels, building_iri FROM points WHERE addr_street IS NOT NULL OR addr_housenumber IS NOT NULL";
```

Comment:
Please move this to line 30 or 32 as all fields should be on top. Either that or create a new static class to store and access this query.